File: java/azure-storage-blob-v8-perfstress/src/main/java/com/azure/storage/blob/v8/perfstress/UploadTest.java
package com.azure.storage.blob.v8.perfstress;

import java.io.IOException;

import com.azure.perfstress.RandomStream;
import com.azure.perfstress.SizeOptions;
import com.azure.storage.blob.v8.perfstress.core.RandomBlobTest;
import com.microsoft.azure.storage.StorageException;

import reactor.core.publisher.Mono;

public class UploadTest extends RandomBlobTest<SizeOptions> {

    public UploadTest(SizeOptions options) {
        super(options);
    }

    @Override
    public void Run() {
        try {
            _cloudBlockBlob.upload(RandomStream.create(Options.Size), Options.Size);
        } catch (StorageException | IOException e) {
            throw new RuntimeException(e);
        }
    }

    @Override
    public Mono<Void> RunAsync() {
        throw new UnsupportedOperationException();
    }
}

File: net/Azure.Storage.Blobs.V11.PerfStress/UploadTest.cs
using Azure.Storage.Blobs.PerfStress.Core;
using Microsoft.Azure.Storage.Blob;
using System;
using System.Threading;
using System.Threading.Tasks;

namespace Azure.Storage.Blobs.PerfStress
{
    public class UploadTest : RandomBlobTest<SizeOptions>
    {
        public UploadTest(SizeOptions options) : base(options)
        { 
        }

        public override void Run(CancellationToken cancellationToken)
        {
            using var stream = RandomStream.Create(Options.Size);
            CloudBlockBlob.UploadFromStream(stream);
        }

        public override async Task RunAsync(CancellationToken cancellationToken)
        {
            using var stream = RandomStream.Create(Options.Size);
            await CloudBlockBlob.UploadFromStreamAsync(stream, cancellationToken);
        }
    }
}
